frontend/sac-mobile/app/(auth)/register.tsx

```diff
@@ -37,114 +65,259 @@ const Register = () => {
+                                value={value}
+                                onSubmitEditing={handleSubmit(onSubmit)}
+                                error={!!errors.id}
+                            />
+                        )}
+                        name="id"
+                        rules={{
+                            required: 'NUID is required',
+                            validate: (value) => {
+                                if (/[^\d]/.test(value)) {
+                                    return 'Please enter a proper NUID number';
+                                }
+                                if (value.length != 9) {
+                                    return 'Please enter 9 digit number';
+                                }
+                                return true;
+                            },
+                        }}
+                    />
+                    {errors.id && <Error message={errors.id.message}/>}
+                </View>
+                <View className="mb-[7%]">
+                    <Controller
+                        control={control}
+                        render={({ field: { onChange, value } }) => (
+                            <DropdownComponent
+                                title="Intended Graduation Year"
+                                item={graduationYear()}
+                                placeholder="Select Year"
+                                onChangeText={onChange}
+                                value={value}
+                                onSubmitEditing={handleSubmit(onSubmit)}
+                                error={!!errors.graduationYear}
+                            />
+                        )}
+                        name="graduationYear"
+                        rules={{ required: 'Graduation year is required' }}
+                    />
+                    {errors.graduationYear && <Error message={errors.graduationYear.message}/>}
+                </View>
+
+                <View className="w-full mb-[8.5%]">
+                    <Controller
+                        control={control}
+                        render={({ field: { onChange, value } }) => (
+                            <Input
+                                title="Password"
+                                autoCorrect={false}
+                                placeholder="Password"
+                                onChangeText={onChange}
+                                value={value}
+                                onSubmitEditing={handleSubmit(onSubmit)}
+                                secureTextEntry={true}
+                                error={!!errors.password}
+                            />
+                        )}
+                        name="password"
+                        rules={{
+                            required: 'Password is required',
+                            validate: (value) => {
+                                if (value.length < 8) {
```

Comment:
our backend password validator checks for 
1. password length
2. presence of a special character
3. presence of a number